pkg/plugin/plugin.go

```diff
@@ -50,3 +55,77 @@ func (p *Plugin) GenerateARM(cs *acsapi.ContainerService) ([]byte, error) {
 func (p *Plugin) HealthCheck(ctx context.Context, cs *acsapi.ContainerService) error {
 	return healthcheck.HealthCheck(ctx, cs)
 }
+
```

Comment:
@openshift/sig-azure ptal in the plugin code.

Reply:
I like it. The only thing I miss is some high-level comment (maybe close to interfaces) on how each flow works. `inplace`, `reimage`, `scale-up scale-down`. Some code level docs on code capabilities and ideas behind.

Reply:
Scaling operations are orthogonal to plugin readiness but I can add some docs about the former.

Reply:
plugin structure looks good to me